### PR TITLE
SONAR-7639 WS api/l10n/index supports only BCP47 language tags

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java [hunk 1]
```diff
@@ -79,10 +80,11 @@ protected void serializeMessages(Request request, Response response) throws IOEx
       Locale locale = userSession.locale();
       String localeParam = request.param("locale");
       if (localeParam != null) {
-        locale = LocaleUtils.toLocale(localeParam);
+        locale = Locale.forLanguageTag(localeParam);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java]: serializeMessages() should be private

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java [hunk 2]
```diff
@@ -79,10 +80,11 @@ protected void serializeMessages(Request request, Response response) throws IOEx
...
+        checkArgument(!locale.getISO3Language().isEmpty(), String.format("'%s' cannot be parsed as a BCP47 language tag", localeParam));
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java]: Use static call for format(), it's more readable

File: server/sonar-server/src/test/java/org/sonar/server/platform/ws/L10nWsTest.java [hunk 3]
```diff
@@ -24,30 +24,27 @@
...
 import org.sonar.server.tester.UserSessionRule;
 import org.sonar.server.ws.WsTester;
 import org.sonar.server.ws.WsTester.Result;
 
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyZeroInteractions;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.class)
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/platform/ws/L10nWsTest.java]: Good move!

**Comment by julienlancelot** [comment]
Very important feedbacks to apply, then LGTM

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java (hunk 1 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java]: ok

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java (hunk 2 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/L10nWs.java]: ok

**Pull request closed without merging**